servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -127,6 +127,12 @@ public void addTempBlockMeta(TempBlockMeta tempBlockMeta)
    */
   public void commitTempBlockMeta(TempBlockMeta tempBlockMeta)
       throws WorkerOutOfSpaceException, BlockAlreadyExistsException, BlockDoesNotExistException {
+    long blockId = tempBlockMeta.getBlockId();
+    if (hasBlockMeta(blockId)) {
```

Comment:
I discussed this with @calvinjia offline and we concluded that the original code makes it possible to create a block with the same id in two different tiers. That is bad because a some functions in block manager assume that the id is unique (e.g. `getBlockMeta(id)` returns BlockMeta for the given id ... it iterate through storage dirs looking for the id until it finds it)

it should not be possible to create a block with the same id in different tiers.